apps/studio/src/pages/home/library/MarketplaceTab/CreateSale.tsx

```diff
@@ -98,13 +99,25 @@ export const CreateSale = () => {
       ),
     totalSaleValue: Yup.number()
       .required("This field is required")
+      .typeError("")
       .when("saleCurrency", {
         is: (val: string) => val === Currency.USD.name,
         otherwise: (s) => s.min(1, "Value must be at least 1 Ɲ"),
         then: (s) => s.min(0.01, "Value must be at least $0.01"),
       }),
   });
 
+  const getDynamicDecimalPrecision = (num: number): number | undefined => {
+    if (num >= 0.001 || num === 0 || !Number.isFinite(num)) {
+      return undefined;
+    }
+    const splitDecimalPortion = num.toFixed(10).split(".")[1];
+    const firstNonZeroIndex = splitDecimalPortion.search(/[1-9]/);
+    return firstNonZeroIndex && firstNonZeroIndex !== -1
+      ? firstNonZeroIndex + 1
+      : undefined;
+  };
```

Comment:
Thinking of refactoring this into crypto or another shared repo utility file. Will add a ticket to take care of refactor as other items in PR are more important to release.

Reply:
Will take a look and create the ticket